vdev/src/git.rs

```diff
@@ -63,10 +63,71 @@ pub fn get_modified_files() -> Result<Vec<String>> {
     Ok(capture_output(&args)?.lines().map(str::to_owned).collect())
 }
 
+pub fn set_config_values(config_values: &[(&str, &str)]) -> Result<String> {
+    let mut args = vec!["config"];
+
+    for (key, value) in config_values {
+        args.push(key);
+        args.push(value);
+    }
+
+    capture_output(&args)
+}
+
+/// Checks if the current directory's repo is clean
+pub fn check_git_repository_clean() -> Result<bool> {
+    Ok(Command::new("git")
+        .args(["diff-index", "--quiet", "HEAD"])
+        .stdout(std::process::Stdio::null())
+        .status()
+        .map(|status| status.success())?)
+}
+
+/// Commits changes from the current repo
+pub fn commit(commit_message: &str) -> Result<Output> {
+    let output = run_command_check_status(&["-am", commit_message])?;
+    Ok(output)
+}
+
+/// Pushes changes from the current repo
+pub fn push() -> Result<Output> {
+    let output = Command::new("git")
+        .arg("push")
+        .output()
+        .map_err(|e| anyhow!("{}", e))?;
+
+    Ok(output)
```

Comment:
The `.map_err` may not be required here:
```suggestion
    Ok(Command::new("git")
        .arg("push")
        .output()?)
```

Reply:
I opted to use the capture_output() function since it was discussed to make everything that uses the output, just use the capture_output() function.